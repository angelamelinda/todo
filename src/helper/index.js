export function mockData() {
    return {
      // example data from redux store
      listTask:{
        
      },
  
      // actions
      RequestAddTask:jest.fn(),
      RequestEditTask:jest.fn(),
      RequestDeleteTask:jest.fn(),
      RequestChangeStatus:jest.fn(),
      RequestGetAllTask:jest.fn()
    }
  }
  