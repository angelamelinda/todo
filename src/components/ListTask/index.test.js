import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { ListTask } from './';
import * as helper from '../../helper';

Enzyme.configure({ adapter: new Adapter()});

describe('ListTask', () => {
    let props = helper.mockData();
    props.isFetched = true;

    let mountedApp;

    const listTask = () => {
        if(!mountedApp) {
            mountedApp = mount (
                <ListTask {...props} />
            )
        }
        return mountedApp;
    }
    beforeEach(() => {
        mountedApp = null
    })

    it('handleDelete when the button delete is clicked', () => {
        let btnDelete = listTask().find('.button-delete');
        expect(btnDelete.exists()).toBe(true);
        btnDelete.first().simulate('click');
        expect(listTask().prop("RequestDeleteTask")).toHaveBeenCalled();
    })

    it('handleEdit when the button edit is clicked', () => {
        props.edit = jest.fn();
        let btnEdit = listTask().find('.button-edit');
        expect(btnEdit.exists()).toBe(true);
        btnEdit.first().simulate('click');
        expect(listTask().prop("edit")).toHaveBeenCalled();
    })

    it('handleChangeStatus when the button edit is clicked', () => {
        let btnChangeStatus = listTask().find('.button-change-status');
        expect(btnChangeStatus.exists()).toBe(true);
        btnChangeStatus.first().simulate('click');
        expect(listTask().prop("RequestEditTask")).toHaveBeenCalled();
    })

    it('matches with snapshot', () => {
        const wrapper = shallow(
            <ListTask {...props} />
        );
        expect(toJson(wrapper),{mode:'shallow'}).toMatchSnapshot();
    })
})

