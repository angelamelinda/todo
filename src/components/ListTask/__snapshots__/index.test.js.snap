// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTask matches with snapshot 1`] = `
<div
  className="list-tasks row"
>
  <div
    className="col-12 task-completed"
    id="5baee5b86d17d7190974e9f9"
    key="0"
  >
    test mock
    <button
      className="button-delete"
      onClick={[Function]}
    >
      Delete
    </button>
    <button
      className="button-edit"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="button-change-status"
      onClick={[Function]}
    >
      Done
    </button>
  </div>
  <div
    className="col-12"
    id="5baeea07e8637f2dd4eb7676"
    key="1"
  >
    maa
    <button
      className="button-delete"
      onClick={[Function]}
    >
      Delete
    </button>
    <button
      className="button-edit"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="button-change-status"
      onClick={[Function]}
    >
      Done
    </button>
  </div>
</div>
`;
